tensorflow/Ubuntu/tensorflow_1.10.0_gpu_ubuntu_16.04.py

```diff
@@ -0,0 +1,210 @@
+    'cp -P /usr/lib/powerpc64le-linux-gnu/libcudnn.so.7.2.1 /usr/local/cuda/lib64',
+    'cp -P /usr/lib/powerpc64le-linux-gnu/libcudnn_static.a /usr/local/cuda/lib64',
+    'cp -P /usr/lib/powerpc64le-linux-gnu/libcudnn_static_v7.a /usr/local/cuda/lib64',
+    ]
+nccl = [
+    'tar -xzvf nccl_2.2.13-1+cuda9.2_ppc64le.tgz',
+    'cp cuda/targets/ppc64le-linux/include/nccl.h /usr/local/cuda/include',
+    'mkdir -p /usr/local/cuda/lib',
+    'cp cuda/targets/ppc64le-linux/lib/libnccl.so /usr/local/cuda/lib/',
+    'cp cuda/targets/ppc64le-linux/lib/libnccl.so.2 /usr/local/cuda/lib/',
+    'cp cuda/targets/ppc64le-linux/lib/libnccl.so.2.2.13 /usr/local/cuda/lib/',
+    'cp cuda/targets/ppc64le-linux/lib/libnccl_static.a /usr/local/cuda/lib/',
+    'chmod a+r /usr/local/cuda/include/',
+    'chmod a+r /usr/local/cuda/lib/',
+    'ldconfig'
+    ]
+bazel = [
+    'mkdir -p bazel',
+    'wget https://github.com/bazelbuild/bazel/releases/download/0.15.0/bazel-0.15.0-dist.zip',
+    'mv bazel-0.15.0-dist.zip bazel/',
+    'unzip -o bazel/bazel-0.15.0-dist.zip -d bazel/',
+    'bazel/compile.sh',
+    'cp bazel/output/bazel /usr/local/bin/'
+    ]
+git = [
+    'rm -rf tensorflow',
+    'git clone -b v1.10.0 https://github.com/tensorflow/tensorflow',
+    'wget  https://github.com/wdirons/tensorflow/commit/03c24d6f13bb30d87745c6eb752019457863b00c.patch',
+    'patch -p1 tensorflow/tensorflow/workspace.bzl 03c24d6f13bb30d87745c6eb752019457863b00c.patch',
+    ]
+
+def run_cmd(command):
+
+    '''
+    Run the given command using check_call and verify its return code.
+    @param str command command to be executed
+    '''
+
+    try:
+        check_call(command.split())
+    except CalledProcessError as e:
+        if command.split()[0] == "rpm": 
+            print('Ignore rpm failure, package is probably already installed')
+        else:
+            print('An exception has occurred: {0}'.format(e))
+            sys.exit(1)
+
+def execute_cmd(list, step):
+
+    '''
+    Execute the given commands using run_cmd function
+    @param list list commands to be executed
+    @param step str name of the comand to be executed
+    '''
+
+    print('Step: %s' % (step))
+    
+    for item in list:
+        run_cmd(item)
+
+def set_environment():
+
+    '''
+    Create bazelrc file with the necessary settings
+    Note: Limiting TF_CUDA_COMPUTE_CAPABILITIES to only
+    GPUs you plan to use will speed build time and decrease
+    overall install size. See https://developer.nvidia.com/cuda-gpus
+    for GPU model to Compute Capabilities mapping.
+    '''
+
+    f= open('tensorflow/.bazelrc',"w+")
+    f.write("build --action_env PYTHON_BIN_PATH='/usr/bin/python'\n\
+build --action_env PYTHON_LIB_PATH='/usr/local/lib/python2.7/site-packages'\n\
+build --python_path='/usr/bin/python'\n\
+build --define with_jemalloc=true\n\
+build --define with_hdfs_support=true\n\
+build:gcp --define with_gcp_support=false\n\
```

Comment:
These build options differ slightly than whats enabled by upstream in their community builds. It also differs a bit from the other recipes in build-scripts. Perhaps we should make them common.

Reply:
I will enable the flags for Google Cloud Platform, Amazon AWS and Apache Kafka. I think that will make things in sync with the upstream defaults.